pkg/virt-launcher/virtwrap/sriov/hostdev.go

```diff
@@ -98,3 +102,88 @@ func decoratePciAddressField(addressField string) (*api.Address, error) {
 	}
 	return decoratedAddrField, nil
 }
+
+type deviceDetacher interface {
+	DetachDevice(xmlData string) error
+}
+
+func DetachHostDevices(dom deviceDetacher, hostDevices []api.HostDevice) error {
+	for _, device := range hostDevices {
+		devXML, err := xml.Marshal(device)
+		if err != nil {
+			return fmt.Errorf("failed to encode (xml) device %v, err: %v", device, err)
+		}
+		err = dom.DetachDevice(string(devXML))
+		if err != nil {
+			return fmt.Errorf("failed to detach device %s, err: %v", devXML, err)
+		}
+		log.Log.Infof("Successfully hot-unplug device: %s (%v)", device.Alias.Name, device.Source.Address)
+	}
+	return nil
+}
+
+func FilterHostDevices(domainSpec *api.DomainSpec) []api.HostDevice {
+	var hostDevices []api.HostDevice
+
+	for _, device := range domainSpec.Devices.HostDevices {
+		if device.Alias != nil && strings.HasPrefix(device.Alias.Name, AliasPrefix) {
+			hostDevices = append(hostDevices, device)
+		}
+	}
```

Comment:
If a SRIOV VM has being created we and old kubevirt version the alis is not going to be there, is this a problem ?